### PR TITLE
feat: optimize skywalking instancename

### LINKED ISSUE #4040: skywalking plugin attribute service_instance_name hard-coded
### Issue description

Whether its attribute “service_instance_name” can be obtained automatically instead of hard-coded in the configuration

### PR DESCRIPTION
**Pull request by dickens7**
Fix #4040

### CODE REVIEW

File: apisix/plugins/skywalking.lua [hunk 1]
```diff
@@ -33,7 +33,7 @@ local metadata_schema = {
         service_instance_name = {
             type = "string",
             description = "User Service Instance Name",
-            default = "APISIX Instance Name",
+            default = core.utils.gethostname(),
```
Comment (spacewander) [inline apisix/plugins/skywalking.lua:36]: We can't use this way as the default value need to be constant

**Review by spacewander — CHANGES_REQUESTED** [review]

File: apisix/plugins/skywalking.lua (hunk 1 above)
Comment (dickens7) [reply apisix/plugins/skywalking.lua:36]: let me see if there are other solutions

Comment (dickens7) [reply apisix/plugins/skywalking.lua]: Use placeholders to implement functionality

File: apisix/plugins/skywalking.lua [hunk 2]
```diff
@@ -126,6 +126,11 @@ function _M.init()
 
     -- TODO: maybe need to fetch them from plugin-metadata
     local metadata_shdict = ngx.shared.tracing_buffer
+
+    if local_plugin_info.service_instance_name == "__hostname__" then
```
Comment (tokers) [inline apisix/plugins/skywalking.lua:130]: The variable style that we use in APISIX always with the prefix `$`. What about `$hostname`.

File: docs/en/latest/plugins/skywalking.md [hunk 3]
```diff
@@ -90,9 +90,9 @@ We can set the endpoint by specified the configuration in `conf/config.yaml`.
 | Name         | Type   | Default  | Description                                                          |
 | ------------ | ------ | -------- | -------------------------------------------------------------------- |
 | service_name | string | "APISIX" | service name for skywalking reporter                                 |
-|service_instance_name|string|"APISIX Instance Name" | service instance name for skywalking reporter |
-| endpoint_addr| string | "http://127.0.0.1:12800" | the http endpoint of Skywalking, for example: http://127.0.0.1:12800 |
-| report_interval| integer | use the value in the skywalking client library | the report interval, in seconds |
+| service_instance_name | string |"APISIX Instance Name" | service instance name for skywalking reporter，  set it to `$hostname` to get local hostname directly.|
```
Comment (spacewander) [inline docs/en/latest/plugins/skywalking.md:93]:
```suggestion
| service_instance_name | string |"APISIX Instance Name" | service instance name for skywalking reporter. If you want to use the machine's hostname directly, set it to `$hostname`.|
```

A little tweak for the doc

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Review by lilien1010 — APPROVED** [review]

**Pull request merged**